### ISSUE TITLE
Ability to resize windowed mode and a fullscreen windowed mode would be nice

### ISSUE THREAD

**Issue by GoatQueerEnjoysChocolate** [Calandiel/SongsOfGPL#141]
This would be great for people who need to alt tab like me

**Comment by Calandiel** [comment]
Fullscreen windowed should be possible (if it's not already a thing).
Resizable windowed mode is more difficult as sote relies on a constant screen resolution to simplify UI code (we have very little manpower behind the project). It'll need to be discussed in more detail.

**Comment by GoatQueerEnjoysChocolate** [comment]
Oh, ok. Because when I altabbed in full windowed mode it crashed, but I forgot to copypaste the crash. Sorry

**Comment by squealingpiggies** [comment]
I know that this has been marked as closed but the master branch only has fullscreen (not even windowed fullscreen) so I took a stab at  resolution changing to understand the code and try my hand at lua for the first time.
I created a simple resolution change/selection in the options that fetches a list of possible resolutions and places it into scrollview as text_buttons that will change an added OPTIONS variable. It also changes the resolutions scaling function to keep its height as a base for drawing units and scales the width to the desired aspect ratio for scaling. This keeps square icons square when changing resolutions and not fullscreen. I had to add an additional check for fullscreen and match the scaling factor to the desktop resolution to keep things square in fullscreen at any resolution. Not the prettiest solution but it works.
I also noticed that the canvas has fixed dimensions and only allocates once. When changing the screen resolution (or just turning on/of fullscreen), the view doesn't resize the canvas and cause the planet to be clipped or leaving black space (that still draws texture icons). I added a check for fullscreen and fix aspect ratio each draw phase but it only really needs to happen when the settings are changed.
![Untitled](https://github.com/Calandiel/SongsOfGPL/assets/158529472/6bb1d177-83fe-4d05-b90a-38c112ba1671)
It is a little messy at time, like changing the resolution while fullscreen will change the size to the new resolution and then fullscreen again. I also think the scrollview is a little buggy since it seems to draw the box/panel too large for the item count (if i wanted 5 items, a value of 4.5 fits perfectly) and clicking buttons inside it sometimes targets the wrong buttons. I still need to look over the drawing code more to figure how to call updates efficiently as every draw phase I recalculate the aspect ratio and check if the window for size change to possibly allocate a new sized canvas when it only needs to do these things on demand.
It is more of a proof of concept but still a possible approach to scaling. I still have to dig through the code some more and find where to place my calls appropriately but is this something that would work, at least in the short term?
